app/Http/Controllers/QuizController.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\StoreQuizResultRequest;
+use App\Models\Quiz;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\Auth;
+
+class QuizController extends Controller
+{
+    public function show(Quiz $quiz): View
+    {
+        $questions = $quiz->questions;
+
+        return view('quizzes.show', [
+            'quiz' => $quiz,
+            'title' => $quiz->title,
+            'questions' => $questions,
+        ]);
+    }
+
+    public function storeQuizResult(StoreQuizResultRequest $request, Quiz $quiz): View
+    {
+        $data = $request->validated();
+        $user = Auth::user();
+
+        $numberOfQuestions = 0;
+        $quizScore = 0;
+
+        foreach ($quiz->questions as $question) {
+            $numberOfQuestions++;
+            $userChoices = $data['question_'.$question->id];
+            if (count($userChoices) < $question->minimum_choices) {
+                continue;
+            } else {
+                $numberOfCorrectAnswers = 0;
+                foreach ($question->choices as $choice) {
+                    if (in_array($choice->value, $userChoices) && $choice->is_answer) {
+                        $numberOfCorrectAnswers++;
+                    }
+                }
```

Comment:
See if this comparison can be done using an array intersection of user choices and correct choices for a given question.

See: https://www.php.net/manual/en/function.array-intersect.php